### PR TITLE
Add always_enable_overscroll property on scrollview

### PR DESCRIPTION
**Pull request by tshirtman**
fix: #6665

### CODE REVIEW

**Comment by matham** [comment]
Shouldn't this be True by default? Also, I think `always_overscroll` is probably better as it's more concise. Although it would be nice to be more specific than `always`, as that's quite vague as to what it means. Maybe even `overscroll_smaller_viewport`, although it's a bit wordy.

**Comment by tshirtman** [comment]
I figured making it True by default could be an unwanted change in behavior, and i tend to be conservative about these things, even if i think it's indeed a better default behavior, if you think it's good enough to allow people who don't want this behavior to disable it, then i'm fine with switching the default value.

the always seems important to me as currently we have overscroll only in the specific situation where the content is big enough to allow scrolling, this just makes the behavior permanent.

Also, it did remind me that it's impossible to overscroll with the mousewheel currently, (and with this patch it's now possible… but only if the content is small enough to prevent normal scrolling :grimacing: :laughing:) i think this should be changed and mouse scroll should work the same as touch scroll in that regard, any opinion?

**Comment by tshirtman** [comment]
Renamed the property, and enabled it by default, documented the fact that you should disable it to get the previous behavior back.

**Comment by matham** [comment]
Looks ok if you're happy with it!?

File: kivy/uix/scrollview.py [hunk 1]
```diff
@@ -276,6 +276,21 @@ def _set_do_scroll(self, value):
...
+    '''Make sure user can overscroll even if there is not enough content
+    to require scrolling.
+
+    This is useful if you want to trigger some action on overscroll, but
+    there is not always enough content to trigger it.
+
+    :attr:`always_overscroll` is a
+    :class:`~kivy.properties.BooleanProperty` and defaults to `True`.
+    versionadded:: 2.0.0
+    The option was added and enabled by default, set to False to get the
```
Comment (pythonic64) [inline kivy/uix/scrollview.py:289]: @tshirtman Shouldn't there be a new line before `versionadded` tag?

Comment (tshirtman) [reply kivy/uix/scrollview.py:289]: hm, maybe, i'll add it along the unittest fix, good call.

Comment (pythonic64) [reply kivy/uix/scrollview.py:289]: Maybe docs won't be generated property without new line.
Also, better readability :)

Comment (tshirtman) [reply kivy/uix/scrollview.py:289]: yes, and the `..` before was missing as well, good thing you mentioned it.

**Comment by idevo89** [comment]
Scrollview and recycleview do not work properly when content size is smaller than parent size.

**always_overscroll=True & content size bigger**
ok

**always_overscroll=True & content size smaller**
Not ok, scrolling down scrolls the scrollview (recycleview) off the screen

**always_overscroll=False & content size bigger**
ok

**always_overscroll=False & content size smaller**
ok untill you resize down so content size is bigger than parent, then you scroll won't work

**To Reproduce**
```python
if __name__ == '__main__':
    from kivy.app import App

    from kivy.uix.gridlayout import GridLayout
    from kivy.uix.button import Button

    class ScrollViewApp(App):

        def build(self):
            layout1 = GridLayout(cols=1, spacing=10, size_hint=(None, None))
            layout1.bind(minimum_height=layout1.setter('height'),
                         minimum_width=layout1.setter('width'))
            for i in range(40):
                btn = Button(text=str(i), size_hint=(None, None),
                             size=(200, 100))
                layout1.add_widget(btn)
            scrollview1 = ScrollView(bar_width='2dp', smooth_scroll_end=10)
            scrollview1.add_widget(layout1)

            layout2 = GridLayout(cols=1, spacing=10, size_hint=(None, None))
            layout2.bind(minimum_height=layout2.setter('height'),
                         minimum_width=layout2.setter('width'))
            for i in range(4):
                btn = Button(text=str(i), size_hint=(None, None),
                             size=(200, 100))
                layout2.add_widget(btn)
            scrollview2 = ScrollView(scroll_type=['bars'],
                                     bar_width='9dp',
                                     scroll_wheel_distance=100)
            scrollview2.add_widget(layout2)

            layout3 = GridLayout(cols=1, spacing=10, size_hint=(None, None))
            layout3.bind(minimum_height=layout3.setter('height'),
                         minimum_width=layout3.setter('width'))
            for i in range(40):
                btn = Button(text=str(i), size_hint=(None, None),
                             size=(200, 100))
                layout3.add_widget(btn)
            scrollview3 = ScrollView(bar_width='2dp', smooth_scroll_end=10,
                                     always_overscroll=False)
            scrollview3.add_widget(layout3)

            layout4 = GridLayout(cols=1, spacing=10, size_hint=(None, None))
            layout4.bind(minimum_height=layout4.setter('height'),
                         minimum_width=layout4.setter('width'))
            for i in range(4):
                btn = Button(text=str(i), size_hint=(None, None),
                             size=(200, 100))
                layout4.add_widget(btn)
            scrollview4 = ScrollView(scroll_type=['bars'],
                                     bar_width='9dp',
                                     scroll_wheel_distance=100,
                                     always_overscroll=False)
            scrollview4.add_widget(layout4)

            root = GridLayout(cols=4)
            root.add_widget(scrollview1)
            root.add_widget(scrollview2)
            root.add_widget(scrollview3)
            root.add_widget(scrollview4)
            return root

    ScrollViewApp().run()
```
